runtime/src/main/java/io/micronaut/logging/LogbackLogLevelConfigurer.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.logging;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import io.micronaut.context.annotation.BootstrapContextCompatible;
+import io.micronaut.context.annotation.Context;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.context.env.Environment;
+import io.micronaut.core.util.StringUtils;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Singleton;
+
+/**
+ * Logback log level configurer.
+ *
+ * @author Denis Stepanov
+ */
+@BootstrapContextCompatible
+@Singleton
+@Context
+@Requires(classes = ch.qos.logback.classic.LoggerContext.class)
+public class LogbackLogLevelConfigurer {
+
+    private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(LogbackLogLevelConfigurer.class);
+
+    /**
+     * Sets log level according to properties.
+     *
+     * @param environment The  environment
+     */
+    public LogbackLogLevelConfigurer(Environment environment) {
```

Comment:
I think the class also needs to implement `ApplicationEventListener<RefreshEvent>` and if `logger.levels` changes apply the changes so logging config can be refreshed at runtime

Reply:
Added but I'm not sure if it's correct.